synapse/handlers/auth.py

```diff
@@ -50,17 +55,16 @@ def __init__(self, hs):
         self.INVALID_TOKEN_HTTP_STATUS = 401
 
         self.ldap_enabled = hs.config.ldap_enabled
-        self.ldap_server = hs.config.ldap_server
-        self.ldap_port = hs.config.ldap_port
-        self.ldap_tls = hs.config.ldap_tls
-        self.ldap_search_base = hs.config.ldap_search_base
-        self.ldap_search_property = hs.config.ldap_search_property
-        self.ldap_email_property = hs.config.ldap_email_property
-        self.ldap_full_name_property = hs.config.ldap_full_name_property
-
-        if self.ldap_enabled is True:
-            import ldap
-            logger.info("Import ldap version: %s", ldap.__version__)
+        if ldap3 and self.ldap_enabled:
```

Comment:
`ldap3` won't be defined if `ldap3` failed to import. I think its probably safe to either a) assume that `ldap3` is a thing given we check in the config, b) set `ldap3 = None`  if we fail to import, c) add a `assert ldap3` line inside the if block, d) check if `ldap3` is defined or e) have a `ldap_imported` bool set in the `try .. except` of the import